python/ray/data/_internal/planner/filter.py

```diff
@@ -1,16 +1,20 @@
-from typing import Callable, Iterator
+from typing import Iterator
 
+from ray.data._internal.compute import BlockTransform, TaskContext
 from ray.data.block import Block, BlockAccessor, RowUDF
 from ray.data.context import DatasetContext
 
 
-def generate_filter_fn() -> Callable[[Iterator[Block]], Iterator[Block]]:
+def generate_filter_fn() -> BlockTransform:
```

Comment:
Shall we define a new type somewhere, e.g. in `execution/operators/map_operator`, instead of referencing `compute/BlockTransform`? We want to get rid of `compute.py` in the future, so let's not add new reference if possible, thanks.